### ISSUE TITLE
Link failure detection [question]

### ISSUE THREAD

**Issue by sunng87** [relayrides/pushy#69]
Hi, we are using a long running pushy 0.3 connection for apple push and it works great by far. But one thing I'm not quite confident about is the link failure detection.

As you may know, there could be a situation when the connection is broken without a FIN sent to client (e.g. net wire unplugged). Then there's no way to know the connection failure on pushy side. Pushy will still hold the ApnsConnection, and Netty's writeAndFlush will always success in both handler and its listener. However, no data will actually be sent to APNS.

Basically, we used heart-beat packet to detect failure in a long-live connection. However, Apple doesn't provide such kind of packet in their protocol. And even worse, there's no response for a success call. So we are totally unaware of the packets' state.

Although we haven't got this issue, but I'm afraid it's possible. So I just want to know if any of you and pushy users had this issue, and how do you deal with that? Do you keep the connection in your application, or just close it after some time ?

Thanks everybody!
